apps/api-runtime/src/main/java/com/akto/runtime/APICatalogSync.java

```diff
@@ -613,6 +615,24 @@ public static boolean isAlphanumericString(String s) {
         return (intCount >= 3 && charCount >= 1);
     }
 
+    private static SubType getTokenSubType(Object token){
+        for (CustomDataType customDataType: SingleTypeInfo.getCustomDataTypesSortedBySensitivity(Context.accountId.get())) {
+            if (!customDataType.isActive()) continue;
+            boolean result = false;
+            Conditions valueConditions =  customDataType.getValueConditions();
+            
+            if(valueConditions !=null){
+                result = valueConditions.validate(token);
+            }
+            if (result) return customDataType.toSubType();
+        }
```

Comment:
We don't need this logic here. Please use `findSubType` directly

Reply:
done